### ISSUE TITLE
[FEATURE]: allow using PVC in container module type

### ISSUE THREAD

**Issue by 10ko** [garden-io/garden#1528]
### Background / Motivation

This keeps coming up from our users.

### What should the user be able to do?

Allow the user to specify a PVC when using the `container` module type.

### Suggested Implementation(s)

To be discussed

### How important is this feature for you/your team?

Crucial (Garden is unusable for us without it)
High (Not having this feature makes using Garden painful)
Medium (Using Garden without it is mildly inconvenient)
Low (Nice to have someday)

**Comment by edvald** [comment]
We had a quick chat internally about this, and my proposal would be to introduce a `kubernetes-container` module type, that is a strict superset of the `container` module type. This will allow us to easily add fields that are very specific to Kubernetes, including specifying PVCs to mount, and also things like Pod annotations (see #1388).

We'd just need to make sure we document its existence well, and provide a usage example.

**Comment by lummie** [comment]
I've been looking at alternatives today, utilising the **kubernetes module** and manifests to create the deployment, service and ingress with a dependency on a **container module** that builds the docker image. 
This nearly works, I get the deployment created in k8s, the service too but I guess there is some `garden-kungfu-magic` happening during a normal container module build as I can't get the ingress to tie up.

Would an alternative to adding PVCs etc to the current container implementation to add a patch-manifest section to the kubernetes module e.g.

```
patches:
  - apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: my-app-container-module-${modules.my-app-container-module.version}
    spec:
      template:
        metadata:
          labels:
            app: my-app
        spec:
          volumes:
            - name: pvc1
              persistentVolumeClaim:
                claimName: volume-claim-name
          containers:
            - name: my-app
              volumeMounts:
                - name: pvc1
                  mountPath: /mnt/pvc1
dependencies: [my-app-container-module]
```

**Comment by eysi09** [comment]
Closed with #1615
